materialsLibrary/src/mix/mixMaterial.ts

```diff
@@ -171,20 +171,48 @@ export class MixMaterial extends PushMaterial {
 
         // Textures
         if (scene.texturesEnabled) {
+            if (this._mixTexture1 && !this._mixTexture1.isReady()) {
+                return false;
+            }
+
+            defines._needUVs = true;
+
+            if (this._mixTexture2) {
+                defines.MIXMAP2 = true;
+            }
+            if (this._mixTexture2 && !this._mixTexture2.isReady()) {
+                return false;
+            }
+
             if (MaterialFlags.DiffuseTextureEnabled) {
-                if (this._mixTexture1) {
-                    if (!this._mixTexture1.isReady()) {
+                if (!this._diffuseTexture1 || !this._diffuseTexture1.isReady()) {
+                    return false;
+                }
+
+                defines.DIFFUSE = true;
+
+                if (!this._diffuseTexture2 || !this._diffuseTexture2.isReady()) {
+                    return false;
+                }
+                if (!this._diffuseTexture3 || !this._diffuseTexture3.isReady()) {
+                    return false;
+                }
+                if (!this._diffuseTexture4 || !this._diffuseTexture4.isReady()) {
+                    return false;
+                }
+
+                if (this._mixTexture2 !== null) {

```

Comment:
Would say:
if (this._mixTexture2) { ... }
In case of users assign « undefined » instead of null

Reply:
Yes you have reasons. I am going to correct.